### PR TITLE
[WIP] Setup Travis

### PR DESCRIPTION
**Pull request by parthsujalshah**
I hve added a .travis.yml file

### CODE REVIEW

**Comment by shubhank-saxena** [comment]
@jsparmani @parthsujalshah the Travis file looks good to me. Reasons that it is failing are not dependent on this PR. 
1. The secret key for django is empty
2. Smoke tests are missing that's why react test won't work

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        yarn: true
+        directories:
+          - node_modeules
+      install:
+        - yarn
+      scripts:
+        - yarn test
+        - yarn build
+    
+    - language: python
```
Comment (shubhank-saxena) [inline .travis.yml:17]: @parthsujalshah split the pre-commit tests and django's run and compile different!
Refer to this 
https://github.com/developer-student-club-thapar/officialWebsite/blob/master/.travis.yml

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      before_install:
+      - pip install poetry
+      install:
+      - poetry install
+      - pre-commit install
+      scripts:
+      - python manage.py makemigrations
+      - python manage.py migrate
+      - python manage.py test
+      - pre-commit run --all-files
```
Comment (shubhank-saxena) [inline .travis.yml:29]: Leave an extra line below the code. I think you are not using pre-commit hooks. Kindly do that!

File: .travis.yml (hunk 1 above)
Comment (shubhank-saxena) [reply .travis.yml:17]: Also, add comments in the codebase!

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+sudo: required
```
Comment (shubhank-saxena) [inline .travis.yml:1]: Make this false

**Comment by jsparmani** [comment]
@parthsujalshah I think you can comment out the js testing script till we have the smoke test for js ready. Other than that, can you `rebase` master onto your branch, so that I can help you setting up the failures with `python` tests due to unavailability of _SECRET_KEY_ .

**Comment by parthsujalshah** [comment]
> @parthsujalshah I think you can comment out the js testing script till we have the smoke test for js ready. Other than that, can you `rebase` master onto your branch, so that I can help you setting up the failures with `python` tests due to unavailability of _SECRET_KEY_ .

Okay

**Pull request closed without merging**